### DISCUSSION TITLE
Ability to mark a comment as optional

### DISCUSSION THREAD

**Discussion by starkej2** [community/community#16278] [Pull Requests]
When reviewing a pull request, I often find myself leaving comments that are optional or nice-to-haves. This type of comment is usually best left to the author's discretion about whether to apply the suggested changes before merging the PR.

GitHub doesn't currently provide any functionality to differentiate between an optional vs required code review comment, so as a workaround my team typically includes prefixes/emoji such as `Optional:`, `Nitpick:`, `💡`, etc to provide this additional context.

It would be really helpful if GitHub pull requests provided a mechanism like an `Optional` checkbox or `Comment Type` dropdown to allow the reviewer to specify whether each individual comment is a required change request or an optional non-blocking request/comment.

**Comment by andreahlert** [comment]
Hey 🙂
This makes a lot of sense, especially for larger PRs where not every note is a blocker. An “optional” or “nit” label could reduce unnecessary tension in reviews.

One question though: would this work best as a per-comment flag or a per-review mode? Some teams might want optional comments by default, while others only occasionally. Native support could standardize what today is just team convention.

**Reply by TastyPi** [reply]
Per comment makes more sense to me, otherwise I would have to do two reviews for "definitely do these" and "maybe do these" instead of all in one review
